engine/movie/opening_cutscene.asm

```diff
@@ -0,0 +1,1235 @@
+	ld a, SPRITE_ANIM_INDEX_GS_INTRO_PIKACHU
+	call InitSpriteAnimStruct
+	depixel 14, 24
+	ld a, SPRITE_ANIM_INDEX_GS_INTRO_PIKACHU_TAIL
+	call InitSpriteAnimStruct
+	ret
+
+IntroScene10:
+; Set up fireball cutscene (evolved Kanto starters)
+	ld hl, wIntroJumptableIndex
+	inc [hl]
+	ld b, SGB_GS_INTRO
+	ld c, 2
+	call GetSGBLayout
+	callab InitEffectObject
+	call Intro_ResetLYOverrides
+	call Intro_BlankTilemapAndBGMap
+
+	ld hl, vChars2
+	ld de, IntroCharizard1GFX
+	lb bc, BANK(IntroCharizard1GFX), 8 tiles
+	call Request2bpp
+
+	ld hl, vFont
+	ld de, IntroCharizard2GFX tile $40
+	lb bc, BANK(IntroCharizard2GFX), 8 tiles
+	call Request2bpp
+
+	ld hl, vChars0
+	ld de, IntroCharizardFlamesGFX
+	lb bc, BANK(IntroCharizardFlamesGFX), 8 tiles
+	call Request2bpp
+
+	ld hl, wSpriteAnimDict
+	ld a, SPRITE_ANIM_INDEX_GS_INTRO_OMANYTE
+	ld [hli], a
+	ld a, SPRITE_ANIM_DICT_DEFAULT
+	ld [hli], a
+
+	ld a, 0
+	call DrawIntroCharizardGraphic
+
+	ld a, $80
+	ldh [hSCY], a
+
+	xor a
+	ldh [hSCX], a
+	ld [wGlobalAnimYOffset], a
+	ld [wGlobalAnimXOffset], a
+
+	xor a
+	ld [wIntroFrameCounter2], a
+	ld a, %00111111
+	ldh [rBGP], a
+	ld a, %11111111
+	ldh [rOBP0], a
+	ret
+
+IntroScene11:
+; scroll up to Charizard silhoutte, flash Blastoise and Venusaur
+	ld hl, wIntroFrameCounter2
+	ld a, [hl]
+	inc [hl]
+	and 1
+	ret z
+	call Intro_CheckSCYEvent
+	ld hl, hSCY
+	ld a, [hl]
+	and a
+	jr z, .next
+	inc [hl]
+	ret
+
+.next
+	ld hl, wIntroJumptableIndex
+	inc [hl]
+	xor a
+	ld [wIntroFrameCounter1], a
+
+IntroScene12:
+; load Charizard palettes
+	ld hl, wIntroFrameCounter1
+	ld a, [hl]
+	inc [hl]
+	srl a
+	srl a
+	and 3
+	ld e, a
+	ld d, 0
+	ld hl, .palettes
+	add hl, de
+	ld a, [hl]
+	and a
+	jr z, .next
+	ldh [rBGP], a
+	ldh [rOBP0], a
+	ret
+
+.next
+	ld hl, wIntroJumptableIndex
+	inc [hl]
+	ld a, $80
+	ld [wIntroFrameCounter1], a
+	ret
+
+.palettes:
+	db %01101010
+	db %10100101
+	db %11100100
+	db %00000000
+
+IntroScene13:
+; Charizard mouth open
+	ld hl, wIntroFrameCounter1
+	ld a, [hl]
+	and a
+	jr z, .next
+
+	dec [hl]
+	ret
+
+.next
+	ld hl, wIntroJumptableIndex
+	inc [hl]
+	ld a, 1
+	call DrawIntroCharizardGraphic
+	ld a, 4
+	ld [wIntroFrameCounter1], a
+	ret
+
+IntroScene14:
+; Charizard breathing fire
+	ld hl, wIntroFrameCounter1
+	ld a, [hl]
+	and a
+	jr z, .next
+	dec [hl]
+	ret
+
+.next
+	ld hl, wIntroJumptableIndex
+	inc [hl]
+	ld a, 2
+	call DrawIntroCharizardGraphic
+	ld a, 64
+	ld [wIntroFrameCounter1], a
+	xor a
+	ld [wIntroFrameCounter2], a
+; fallthrough
+
+IntroScene15:
+; Charizard mouth wide open / fireball starts
+	call Intro_AnimateFireball
+	ld hl, wIntroFrameCounter1
+	ld a, [hl]
+	and a
+	jr z, .next
+	dec [hl]
+	ret
+.next
+	ld hl, wIntroJumptableIndex
+	inc [hl]
+	xor a
+	ld [wIntroFrameCounter1], a
+	ret
+
+IntroScene16:
+; continue fireball / fade out palettes
+	call Intro_AnimateFireball
+	ld hl, wIntroFrameCounter1
+	ld a, [hl]
+	inc [hl]
+	swap a
+	and $07
+	ld e, a
+	ld d, $00
+	ld hl, .palettes
+	add hl, de
+	ld a, [hl]
+	cp $ff
```

Comment:
`cp -1` since the `.palettes` data below ends with standard `-1` (and a blank line between `ret` and `.palettes:`)